clippy_lints_internal/Cargo.toml

```diff
@@ -1,7 +1,7 @@
 [package]
 name = "clippy_lints_internal"
 version = "0.0.1"
-edition = "2021"
+edition = "2024"
```

Comment:
Is it supposed to be here? (once again, it can, but it should be in a separate commit)

Reply:
Yeah so that `#![feature(let_chains)]` can be removed

Reply:
Would you mind putting it in a first commit? In both cases feel free to r=me.